cmd/gapit/benchmark.go

```diff
@@ -17,921 +17,1234 @@ package main
+			if err != nil {
+				return nil, err
+			}
+			if rp.children == nil {
+				if _, err := b.measure(ctx, expandCommandTreeNode(rp)); err != nil {
+					return nil, err
+				}
+			}
 
-	m, err := client.Get(ctx, c.Messages().Path(), nil)
-	if err != nil {
-		return err
+			level := 4
+			parent := rp
+			if secondary {
+				level = 6
+				exec, err := rp.chooseExecute(choose("execute", 4))
+				if err != nil {
+					return nil, err
+				}
+				if exec.children == nil {
+					if _, err := b.measure(ctx, expandCommandTreeNode(exec)); err != nil {
+						return nil, err
+					}
+				}
+
+				parent, err = exec.chooseCommandBuffer(choose("secondary command buffer", 5))
+				if err != nil {
+					return nil, err
+				}
+				if parent.children == nil {
+					if _, err := b.measure(ctx, expandCommandTreeNode(parent)); err != nil {
+						return nil, err
+					}
+				}
+			}
+
+			return parent.chooseDrawCall(choose("draw call", level))
+		},
 	}
-	messages := m.(*service.Messages)
+}
 
-	boxedVal, err := client.Get(ctx, (&path.Stats{
-		Capture:  c,
-		DrawCall: false,
-	}).Path(), nil)
-	if err != nil {
-		return err
+func loadAttachments(after *path.Command) action {
+	return &simple{
+		"attachments",
+		func(ctx context.Context, b *benchmark) (interface{}, error) {
+			return b.client.Get(ctx, (&path.FramebufferAttachments{
+				After: after,
+			}).Path(), b.resolveConfig())
+		},
 	}
-	traceStartTimestamp := boxedVal.(*service.Stats).TraceStart
-
-	frameTimes := []uint64{}
-
-	stateBuildTime := int64(0)
-	stateBuildStartTime := traceStartTimestamp
-	stateBuildEndTime := traceStartTimestamp
-	hasStateSerialization := false
-	frameRe := regexp.MustCompile("Frame Number: [\\d]*")
-	for _, m := range messages.List {
-		if m.Message == "State serialization started" {
-			hasStateSerialization = true
-			stateBuildStartTime = m.Timestamp
-		} else if m.Message == "State serialization finished" {
-			stateBuildEndTime = m.Timestamp
-			stateBuildTime = int64(stateBuildEndTime - stateBuildStartTime)
-		} else if !hasStateSerialization && frameRe.MatchString(m.Message) {
-			frameTimes = append(frameTimes, m.Timestamp)
-		}
+}
+
+func loadAttachment(after *path.Command, index uint32) action {
+	return &simple{
+		fmt.Sprintf("attachment<%d>", index),
+		func(ctx context.Context, b *benchmark) (interface{}, error) {
+			return b.client.Get(ctx, (&path.FramebufferAttachment{
+				After:          after,
+				Index:          index,
+				RenderSettings: fbRenderSettings,
+				Hints:          fbHints,
+			}).Path(), b.resolveConfig())
+		},
 	}
+}
+
+func loadFrameBuffer(after *path.Command) action {
+	return &simple{
+		"framebuffer",
+		func(ctx context.Context, b *benchmark) (interface{}, error) {
+			attsBoxed, err := b.measure(ctx, loadAttachments(after))
+			if err != nil {
+				return nil, err
+			}
+			atts := attsBoxed.(*service.FramebufferAttachments).GetAttachments()
+			// todo: load thumbnails
```

Comment:
Did you meant to do this before landing this PR?

Reply:
Removed the TODO.